bossdata/meta.py

```diff
@@ -98,36 +100,74 @@ def create_meta_lite(sp_all_path, db_path, verbose=True):
+
+    # Using ascii.fixed_width_two_line format, but still have to specify
+    # skipping the first two lines when reading by chunks.
+    init_position = 2
+    position = init_position
+
+    '''
+    Things that have been tried RE: speeding this up to no (or very little) effect:
+    1.)  Removing the primary key from the table creation, doing all insets, and
+        then running SQL:
+            CREATE UNIQUE INDEX psuedo_primary_key_meta ON meta (PLATE,MJD,FIBER)
+        at very end.
+    2.)  PRAGMA synchronous = OFF
+        PRAGMA journal_mode = OFF
+        and variations
+    3.)  Explicitly setting transactions (which fails with an error, as this is
+        being done implicitly)
+
+    Currently, the time split for each chunk of rows is about 12/58 : parsing/inserting
+    '''
+
     # Read the database into memory.
     if verbose:
         print('Initializing the lite database...')
+
     with gzip.open(sp_all_path, mode='r') as f:
-        table = astropy.table.Table.read(f, format='ascii')
-
-    # Create a new database file.
-    rules = {}
-    for i in range(5):
-        rules['MODELFLUX{}'.format(i)] = 'MODELFLUX_{}'.format(i)
-    sql, num_cols = sql_create_table(
-        'meta', table.dtype, renaming_rules=rules, primary_key='(PLATE,MJD,FIBER)')
-    connection = sqlite3.connect(db_path)
-    cursor = connection.cursor()
-    cursor.execute(sql)
+        table = astropy.table.Table.read(f, data_start=position,
+                                         data_end=position+chunk_size,
+                                         format='ascii.fixed_width_two_line',
+                                         guess=False)
+
+        # Create a new database file.
+        rules = {}
+        for i in range(5):
+            rules['MODELFLUX{}'.format(i)] = 'MODELFLUX_{}'.format(i)
+        sql, num_cols = sql_create_table(
+            'meta', table.dtype, renaming_rules=rules, primary_key='(PLATE,MJD,FIBER)')
+        connection = sqlite3.connect(db_path)
+        cursor = connection.cursor()
+        cursor.execute(sql)
+        connection.commit()
+
+        # Insert rows into the database.
+        sql = 'INSERT INTO meta VALUES ({values})'.format(values=','.join('?' * num_cols))
 
-    # Insert rows into the database.
-    sql = 'INSERT INTO meta VALUES ({})'.format(','.join('?' * num_cols))
-    if verbose:
-        progress_bar = ProgressBar(
-            widgets=['Writing', ' ', Percentage(), Bar()], maxval=len(table)).start()
-    for i, row in enumerate(table):
-        cursor.execute(sql, row)
         if verbose:
-            progress_bar.update(i + 1)
-    connection.commit()
-    connection.close()
-    if verbose:
-        progress_bar.finish()
+            # Don't know final count, can't do percentage, bar, etc. Display work done instead
+            widgets = [FormatLabel('Processed: %(value)d lines (in: %(elapsed)s)')]
+            progress_bar = ProgressBar(widgets=widgets, maxval=sys.maxint).start()
+
```

Comment:
It is just "a big number".  I need to have some value otherwise progress_bar.update throws an error.

I was thinking, however, of a lower-level way to get around this; since we know the input file format is fixed, know the byte length per line, it just simple division to get the row count if the total size of the file is known.

Missing finish() was a mistake, not intentional.